jenkins/pipelines/ci/tiflash/tiflash-ghpr-integration-tests.groovy

```diff
@@ -0,0 +1,166 @@
+                    disableSubmodules   : true,
+                    parentCredentials   : true,
+                    recursiveSubmodules : true,
+                    trackingSubmodules  : false,
+                    reference           : ''],
+                    [$class: 'PruneStaleBranch'],
+                            [$class: 'CleanBeforeCheckout'],
+                    ],
+                    doGenerateSubmoduleConfigurations: false,
+                ])
+        }
+        dir('tests/.build') {
+            stage('Get Artifacts') {
+                copyArtifacts(
+                    projectName: 'tiflash-build-common',
+                    parameters: "TARGET_BRANCH=${ghprbTargetBranch},TARGET_PULL_REQUEST=${ghprbPullId},TARGET_COMMIT_HASH=${ghprbActualCommit},BUILD_TIFLASH=true",
+                    selector: lastSuccessful(),
+                    optional: false)
+                sh "tar -zxvf tiflash.tar.gz"
+            }
+        }
+
+        stage('Stash') {
+            stash name: "code-and-artifacts", useDefaultExcludes: true
+        }
+    }
+}
+
+node(GO_TEST_SLAVE) {
+    def toolchain = null
+    def identifier = "tiflash-integration-test-${ghprbTargetBranch}-${ghprbPullId}"
+    def repo_path = "/home/jenkins/agent/workspace/tiflash-build-common/tics"
+    def tidb_branch = ({
+        def m = ghprbCommentBody =~ /tidb\s*=\s*([^\s\\]+)(\s|\\|$)/
+        if (m) {
+            return "${m.group(1)}"
+        }
+        return ghprbTargetBranch ?: 'master'
+    }).call()
+
+    stage('Wait for Build') {
+        waitUntil {
+            def api = "https://ci.pingcap.net/job/tiflash-build-common/api/xml?tree=allBuilds[result,building,actions[parameters[name,value]]]&xpath=(//allBuild[action[parameter[name=%22TARGET_COMMIT_HASH%22%20and%20value=%22${ghprbActualCommit}%22]%20and%20parameter[name=%22BUILD_TIFLASH%22%20and%20value=%22true%22]]])[1]"
+            def response = httpRequest api
+            def content = response.getContent()
+            if (content.contains('<result>FAILURE</result>')) {
+                error "build failure"
+            }
+            if (!content.contains('<building>false</building>') || !content.contains('<result>SUCCESS</result>')) {
+                return false
+            }
+            return true
+        }
+        copyArtifacts(
+            projectName: 'tiflash-build-common',
+            parameters: "TARGET_BRANCH=${ghprbTargetBranch},TARGET_PULL_REQUEST=${ghprbPullId},TARGET_COMMIT_HASH=${ghprbActualCommit},BUILD_TIFLASH=true",
+            filter: 'toolchain',
+            selector: lastSuccessful(),
+            optional: false
+        )
+        toolchain = readFile(file: 'toolchain').trim()
+        echo "Built with ${toolchain}"
+    }
+
+    checkoutTiFlash()
+
+    parallel (
+        "tidb ci test": {
+            def name = "tidb-ci-test"
+            runTest("${identifier}-${name}", name, "tests/tidb-ci", tidb_branch)
```

Comment:
pod lable add ${BUILD_NUMBER} to avoid direct impact of different builds

example "${identifier}-${name}-${BUILD_NUMBER}"